### PR TITLE
Example_ynh

### PR DESCRIPTION
**Pull request by ericgaspar**

### CODE REVIEW

**Comment by ericgaspar** [comment]
!testme

**Comment by ericgaspar** [comment]
!testme

File: scripts/upgrade [hunk 1]
```diff
@@ -3,275 +3,206 @@
...
-;&
-"1.13."* )
-    ynh_setup_source $final_path source/${architecture}_1.14
-    restart_gitea
-;&
-"1.14."* )
-    ynh_setup_source $final_path source/${architecture}_1.15
-    restart_gitea
-;&
-esac
```
Comment (Josue-T) [inline scripts/upgrade:210]: As sad here https://github.com/YunoHost-Apps/gitea_ynh/pull/45#discussion_r884315843 you can't remove this.

**Review by Josue-T — CHANGES_REQUESTED** [review]
And can you add the auto upgrade script based on this with some  improvement and adaptation (it is the actual script that I use to upgrade gitea)
```bash
     get_checksum() {
            local v=$1
           wget https://github.com/go-gitea/gitea/releases/download/v${v}/gitea-${v}-linux-arm-6.sha256
        #     wget https://github.com/go-gitea/gitea/releases/download/v${v}/gitea-${v}-linux-arm-7.sha256
            wget https://github.com/go-gitea/gitea/releases/download/v${v}/gitea-${v}-linux-386.sha256
            wget https://github.com/go-gitea/gitea/releases/download/v${v}/gitea-${v}-linux-amd64.sha256
            wget https://github.com/go-gitea/gitea/releases/download/v${v}/gitea-${v}-linux-arm64.sha256
        
            sha256sumarm6=$(cat gitea-${v}-linux-arm-6.sha256 | cut -d' ' -f1)
            sha256sumarm7=$(cat gitea-${v}-linux-arm-6.sha256 | cut -d' ' -f1)
            sha256sum386=$(cat gitea-${v}-linux-386.sha256 | cut -d' ' -f1)
            sha256sumamd64=$(cat gitea-${v}-linux-amd64.sha256 | cut -d' ' -f1)
            sha256sumarm64=$(cat gitea-${v}-linux-arm64.sha256 | cut -d' ' -f1)

            rm gitea-${v}-linux-arm-6.sha256
        #     rm gitea-${v}-linux-arm-7.sha256
            rm gitea-${v}-linux-386.sha256
            rm gitea-${v}-linux-amd64.sha256
            rm gitea-${v}-linux-arm64.sha256
        }
        update_source_file() {
            local suffix=$1
            local v=$2
            # 	Update this link
        # 	SOURCE_URL=https://github.com/go-gitea/gitea/releases/download/v__APP_VERSION__/gitea-v__APP_VERSION__
            sed -r -i "s@SOURCE_URL=(.*)/releases/download/v[[:alnum:].]{4,8}/gitea-[[:alnum:].]{4,8}-linux-arm-6@SOURCE_URL=\1/releases/download/v${v}/gitea-${v}-linux-arm-6@" conf/source/arm${suffix}
            sed -r -i "s@SOURCE_SUM=[[:alnum:]]{64}@SOURCE_SUM=$sha256sumarm6@" conf/source/arm${suffix}
            sed -r -i "s@SOURCE_URL=(.*)/releases/download/v[[:alnum:].]{4,8}/gitea-[[:alnum:].]{4,8}-linux-arm-6@SOURCE_URL=\1/releases/download/v${v}/gitea-${v}-linux-arm-6@" conf/source/armv7${suffix}
            sed -r -i "s@SOURCE_SUM=[[:alnum:]]{64}@SOURCE_SUM=$sha256sumarm7@" conf/source/armv7${suffix}
            sed -r -i "s@SOURCE_URL=(.*)/releases/download/v[[:alnum:].]{4,8}/gitea-[[:alnum:].]{4,8}-linux-386@SOURCE_URL=\1/releases/download/v${v}/gitea-${v}-linux-386@" conf/source/i386${suffix}
            sed -r -i "s@SOURCE_SUM=[[:alnum:]]{64}@SOURCE_SUM=$sha256sum386@" conf/source/i386${suffix}
            sed -r -i "s@SOURCE_URL=(.*)/releases/download/v[[:alnum:].]{4,8}/gitea-[[:alnum:].]{4,8}-linux-amd64@SOURCE_URL=\1/releases/download/v${v}/gitea-${v}-linux-amd64@" conf/source/x86-64${suffix}
            sed -r -i "s@SOURCE_SUM=[[:alnum:]]{64}@SOURCE_SUM=$sha256sumamd64@" conf/source/x86-64${suffix}
            sed -r -i "s@SOURCE_URL=(.*)/releases/download/v[[:alnum:].]{4,8}/gitea-[[:alnum:].]{4,8}-linux-arm64@SOURCE_URL=\1/releases/download/v${v}/gitea-${v}-linux-arm64@" conf/source/arm64${suffix}
            sed -r -i "s@SOURCE_SUM=[[:alnum:]]{64}@SOURCE_SUM=$sha256sumarm64@" conf/source/arm64${suffix}
        }
        
        pushd repos/gitea
        
        git checkout auto_update
        git pull -f gitea auto_update:auto_update

        last_version=$(egrep '\"version\": \"[[:alnum:].]{4,8}~ynh([[:digit:]]+)\",' manifest.json | cut -d'~' -f1 | egrep -o '[[:alnum:].]{4,8}$')
        
        last_main_version=${last_version%.*}
        main_version=${gitea_version%.*}
        
        if [ $last_main_version != $main_version ]; then
            cp conf/source/arm.src conf/source/arm_${last_main_version}.src
            cp conf/source/armv7.src conf/source/armv7_${last_main_version}.src
            cp conf/source/i386.src conf/source/i386_${last_main_version}.src
            cp conf/source/x86-64.src conf/source/x86-64_${last_main_version}.src
            cp conf/source/arm64.src conf/source/arm64_${last_main_version}.src

            last_main_version_full=$(curl 'https://api.github.com/repos/go-gitea/gitea/releases' -H 'Host: api.github.com' --compressed \
                | grep '"tag_name":' | grep -o -P '(\d+\.)*\d+' | grep "^$last_main_version" | head -n1)

            get_checksum $last_main_version_full
            update_source_file "_$last_main_version.src" $last_main_version_full
    #         WE need to add this
    #         "1.7."* )
    #             ynh_setup_source $final_path source/${architecture}_1.8
    #             restart_gitea
    #         ;&
    #         esac
            last_last_version=$(grep 'ynh_setup_source $final_path source/${architecture}_' scripts/upgrade \
                | tail -n1 | egrep -o '[[:alnum:]]+\.[[:alnum:]]+$')
            perl -0777 -pe "s@restart_gitea\n;&\nesac@restart_gitea\n;&\n\"${last_last_version}."'"* )\n    ynh_setup_source \$final_path source/\${architecture}_'"${last_main_version}\n    restart_gitea\n;&\nesac@" scripts/upgrade > tmp
            mv tmp scripts/upgrade
        fi
    
        get_checksum $gitea_version
        update_source_file ".src" $gitea_version
        sed -r -i "s@\"version\": \"[[:alnum:].]{4,8}~ynh([[:digit:]]+)\",@\"version\": \"${gitea_version}~ynh1\",@" manifest.json
        sed -r -i "s@\*\*Shipped version:\*\* [[:alnum:].]{4,8}@\*\*Shipped version:\*\* ${gitea_version}@" README.md

        git add -A
        git commit -a -m "Upgrade gitea to $gitea_version"
        git push gitea auto_update:auto_update
```

File: scripts/upgrade (hunk 1 above)
Comment (ericgaspar) [reply scripts/upgrade:210]: ok. Closing then

**Pull request closed without merging**